component/src/main/java/org/wso2/extension/siddhi/execution/rdbms/QueryStreamProcessor.java

```diff
@@ -0,0 +1,253 @@
+        this.siddhiAppName = siddhiAppContext.getName();
+
+        if ((attributeExpressionExecutors.length != 3)) {
+            throw new SiddhiAppValidationException(this.siddhiAppName + " - rdbms query function " +
+                    "should have 3 parameters , but found '" + attributeExpressionExecutors.length + "' parameters.");
+        }
+
+        this.dataSourceName = RDBMSStreamProcessorUtil.validateDatasourceName(attributeExpressionExecutors[0],
+                this.siddhiAppName);
+
+        if (attributeExpressionExecutors[1].getReturnType() == Attribute.Type.STRING) {
+            queryExpressionExecutor = attributeExpressionExecutors[1];
+        } else {
+            throw new SiddhiAppValidationException(this.siddhiAppName + " - The parameter 'query' in rdbms query " +
+                    "function should be of type STRING, but found a parameter with type '" +
+                    attributeExpressionExecutors[1].getReturnType() + "'.");
+        }
+
+        if (attributeExpressionExecutors[2] instanceof ConstantExpressionExecutor) {
+            String attributeDefinition = ((ConstantExpressionExecutor) attributeExpressionExecutors[2])
+                    .getValue().toString();
+            this.attributeList = Arrays.stream(attributeDefinition.split(","))
+                    .map((String attributeExp) -> {
+                        String[] splitAttributeDef = attributeExp.trim().split("\\s");
+                        if (splitAttributeDef.length != 2) {
+                            throw new SiddhiAppValidationException(this.siddhiAppName + " - The parameter " +
+                                    "'attribute.definition.list' is invalid, it should be comma separated list of " +
+                                    "<AttributeName AttributeType>, but found '" + attributeDefinition);
+                        }
+                        Attribute.Type attributeType;
+                        switch (splitAttributeDef[1].toLowerCase()) {
+                            case "bool":
+                                attributeType = Attribute.Type.BOOL;
+                                break;
+                            case "double":
+                                attributeType = Attribute.Type.DOUBLE;
+                                break;
+                            case "float":
+                                attributeType = Attribute.Type.FLOAT;
+                                break;
+                            case "int":
+                                attributeType = Attribute.Type.INT;
+                                break;
+                            case "long":
+                                attributeType = Attribute.Type.LONG;
+                                break;
+                            case "string":
+                                attributeType = Attribute.Type.STRING;
+                                break;
+                            default:
```

Comment:
Do we not support object type in Siddhi?

Reply:
As of now, we are not supporting Siddhi object data type through this extension.

Reply:
I think we had to have object only for Thrift purposes, so it's fine to not support object types here.